src/commands/generate/index.ts

```diff
@@ -1,15 +1,37 @@
 import { Command } from '@oclif/core';
-import * as inquirer from 'inquirer';
+import { prompt } from 'inquirer';
 
-import Advanced from '../../templates/aws/advanced';
+import { getTargetDir } from '../../helpers/file';
+import { detectTerraform, formatCode } from '../../helpers/terraform';
+import { generateAwsTemplate } from '../../templates/aws';
 
-type GenerateOption = {
+type GeneralOptions = {
   projectName: string;
-  platform: string;
-  infrastructureType: string;
-  awsRegion: string;
+  provider: string;
 };
 
+const providerChoices = [
+  {
+    type: 'list',
+    name: 'provider',
+    message: 'Which cloud provider would you like to use?',
+    choices: [
+      {
+        value: 'aws',
+        name: 'AWS',
+      },
+      {
+        value: 'gcp',
+        name: 'GCP',
```

Comment:
```suggestion
        name: 'GCP (NOT IMPLEMENTED YET)',
```
Same for Heroku 🙇 
It was quite frustrating when I tried this template and had to start the whole command again as I selected a non implemented option :D

Reply:
🙇 Added in https://github.com/nimblehq/infrastructure-templates/pull/70/commits/171467de4b5e65dfede751df80398d1d6f33ed91
I also disable those options 😄